lib/portlayer/storage/archiver.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"io"
+	"net/url"
+
+	"github.com/vmware/vic/pkg/trace"
+)
+
+// Archiver defines an API for creating archives consisting of data that exists between a child and parent
+// layer
+type Archiver interface {
+
+	// Read reads the delta between child and parent image layers, returning
+	// the difference as a tar archive.
+	//
+	// store - the image store containing the two layers
+	// child - the leaf layer of the image
+	// ancestor - the layer up the chain against which to diff
+	// pathspec - describes filters on paths found in the data (include, exclude, strip)
+	// data - set to true to include file data in the tar archive, false to include headers only
+	Read(op trace.Operation, store *url.URL, child, ancestor string, pathspec map[string]string, data bool) (io.ReadCloser, error)
```

Comment:
This will actually also have a `Write` method as well, but `Archiver` follows that -er suffix pattern already (if that's what you mean). Maybe we should rename `Read` to `ReadArchive`?